File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_industrial_digital_in_4'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
idi4 = BrickletIndustrialDigitalIn4.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current value as bitmask
value_mask = idi4.get_value
puts "Value Mask: #{'%04b' % value_mask}"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "xyz" # Change to your UID

import time

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_industrial_digital_in_4 import IndustrialDigitalIn4

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    idi4 = IndustrialDigitalIn4(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Read out values as bitmask
    value = idi4.get_value()
    print('Value: ' + str(bin(value)))

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()
